mantidproject/mantid
# Use DetectorGroupingOptions widget in diffraction interface

### Description of work
This PR removes the old grouping options in the Indirect Diffraction interface and replaces them with the DetectorGroupingOptions widget used in the Energy Transfer tab of Indirect Data Reduction. This has several extra grouping features including IDF, Individual, File and Custom grouping.

These grouping options are now available for:
- diffspec mode: IRIS, OSIRIS, TOSCA and VESUVIO
- diffonly mode: IRIS, TOSCA and VESUVIO

Fixes #36764

### To test:

1. Open `Indirect`->`Diffraction`
2. Select `IRIS` instrument
3. The default detector grouping should be `All`
4. Load run number 26176
5. Click `Run`
6. The output workspace should have 1 spectra
7. Choose `Individual`, click `Run`, the output workspace should have 8 spectra
8. Choose `Groups`, select 4 groups, click `Run`, the output workspace should have 4 spectra
9. Choose `Custom`, enter `105-108,109-112`, click `Run`, the output workspace should have 2 spectra
10. Still using `Custom`, try spectra numbers outside of the range 105-112. This should not be allowed
11. Using `Groups`, try more than 8 groups. This should not be allowed
12. Use an empty `Custom` grouping string. This should not be allowed
13. Use an empty `File` string. This should not be allowed



### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
